testing/public-api-tests/src/integTest/groovy/org/gradle/api/PublicApiIntegrationTest.groovy

```diff
@@ -138,10 +138,11 @@ class PublicApiIntegrationTest extends AbstractIntegrationSpec implements JavaTo
 
             import org.gradle.api.Plugin
             import org.gradle.api.Project
+            import org.gradle.kotlin.dsl.*
 
             class PublishedApiTestPlugin : Plugin<Project> {
                 override fun apply(project: Project) {
-                    project.tasks.register("myTask", CustomTask::class.java) { task ->
+                    val myTask = project.tasks.registering(CustomTask::class) { task ->
```

Comment:
💅 is this `val myTask` needed?

Reply:
Where would the task name come from otherwise?

Reply:
Oh, it's needed for task name